kubernetes/cache/cache.go

```diff
@@ -149,6 +151,34 @@ func NewKialiCache() (KialiCache, error) {
 	kialiCacheImpl.k8sApi = istioClient.GetK8sApi()
 	kialiCacheImpl.istioApi = istioClient.Istio()
 
+	// Update SA Token
+	ticker := time.NewTicker(tokenExpireDuration)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if newToken, err := kubernetes.GetKialiToken(); err != nil {
+					log.Errorf("Error updating Kiali Token")
+				} else {
+					log.Debug("Kiali Cache: Updating token")
+					istioConfig.BearerToken = newToken
+					var errorInitClient error
+					istioClient, errorInitClient = kubernetes.NewClientFromConfig(&istioConfig)
+					if errorInitClient != nil {
+						log.Errorf("Error creating new Client From Config")
```

Comment:
How about also logging with error itself, since you got the return error? Then the error logs make more sense, it is better for us to find the reason why it failed.

Reply:
Good idea! Thanks